pkg/action/action_test.go

```diff
@@ -301,3 +308,63 @@ func TestGetVersionSet(t *testing.T) {
 		t.Error("Non-existent version is reported found.")
 	}
 }
+
+func TestGetCapabilities(t *testing.T) {
+	t.Run("Capabilities.KubeVersion gives the correct version information", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		mockDiscoveryClient := discovery.NewMockCachedDiscoveryInterface(ctrl)
+		serverVersion := &version.Info{
+			GitVersion: "v1.16.0",
+			Major:      "1",
+			Minor:      "16",
+			GoVersion:  runtime.Version(),
+			Compiler:   runtime.Compiler,
+			Platform:   fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH),
+		}
+		mockDiscoveryClient.EXPECT().Invalidate()
+		mockDiscoveryClient.EXPECT().ServerVersion().Return(serverVersion, nil)
+		mockDiscoveryClient.EXPECT().ServerGroupsAndResources().Return(nil, nil, nil)
+
+		mockRESTClientGetter := NewMockRESTClientGetter(ctrl)
+
+		mockRESTClientGetter.EXPECT().ToDiscoveryClient().Return(mockDiscoveryClient, nil)
+
+		configuration := &Configuration{RESTClientGetter: mockRESTClientGetter}
```

Comment:
Aren't we just testing the behaviour that the mock discovery client responds with the expected output, rather than testing the actual discovery client? I'm having a hard time seeing how this tests the discovery client.

Reply:
yeah, so stub. let me see if we can do better. I can't remember much now why this was done 🙈 I think this was the only way that I could find?

Reply:
I need to be able to stub the server right? so when the code calls the server, the stub is used to get the information. just checking if a call was made is not enough. that would mean just verifying if a call was made. This is more around, when the server has a given information, the same is shown by the code that fetches the information and uses it

Reply:
I'm only making sure the mock discover client returns the dummy information that the server has and then I test how the code interprets the information. I'm not testing the discovery client. The subject under test here is `getCapabilities` method here, and I'm stubbing the external calls. If I was testing discovery client, the subject under test, then mocking it would be wrong 🙈

Reply:
does that make sense?